test/test_xpu.py

```diff
@@ -519,6 +520,42 @@ def test_device_memory_allocated(self):
         )
         del a
 
+    def test_memory_stats(self):
```

Comment:
Nothing runs in parallel of this test nor anything is going to jitter this and will make this test flaky?

Reply:
Some tests run in other threads using pytest iirc

Reply:
Let me enhance this UT a bit.